src/metaswitch/clearwater/etcd_shared/common_etcd_synchronizer.py

```diff
@@ -120,6 +122,129 @@ def _error_catcher(self): # pragma: no cover
             self.release_conn()
 urllib3.HTTPResponse._error_catcher = _error_catcher
 
+# Monkeypatch python-etcd to catch MaxRetryErrors. This is a hacky fix to cover
+# https://github.com/jplana/python-etcd/issues/160 (it doesn't cover why
+# urllib3 is hitting a MaxRetryError in the first place). We should remove this
+# if we update the python-etcd version
+def _patched_wrap_request(payload): # pragma: no cover
```

Comment:
Minor point: I got a little confused about the use of "payload" here (was confusing it with an HTTP payload, like "json_payload" below). I'd have called it something else that indicates more clearly that it's a function that's being wrapped, but maybe that's just because I've not used functools.wraps much before.